OmsAgent/omsagent.py

```diff
@@ -46,100 +49,279 @@
-                hutil = parse_context("Disable")
-                disable(hutil)
+                operation = 'Disable'
             elif re.match("^([-/]*)(uninstall)", a):
-                hutil = parse_context("Uninstall")
-                uninstall(hutil)
+                operation = 'Uninstall'
             elif re.match("^([-/]*)(install)", a):
-                hutil = parse_context("Install")
-                install(hutil)
+                operation = 'Install'
             elif re.match("^([-/]*)(enable)", a):
-                hutil = parse_context("Enable")
-                enable(hutil)
+                operation = 'Enable'
             elif re.match("^([-/]*)(update)", a):
-                dummy_command("Update", "success", "Update succeeded")
+                operation = 'Update'
     except Exception as e:
-        err_msg = ("Failed to enable the extension with error: {0}, "
-                   "{1}").format(e, traceback.format_exc())
-        waagent.Error(err_msg)
-        hutil.error(err_msg)
-        hutil.do_exit(1, 'Enable','failed','0',
-                      'Enable failed: {0}'.format(e))
+        waagent.Error(e.message)
+
+    # Set up for exit code and any error messages
+    exit_code = 0
+    message = operation + ' succeeded'
+
+    # Invoke operation
+    try:
+        hutil = parse_context(operation)
+        if is_vm_supported_for_extension():
+            exit_code = operations[operation](hutil)
+        else:
+            log_and_exit(hutil, operation, 51, 'Unsupported operation system')
+
+        if exit_code is not 0:
+            message = (operation + ' failed with exit code {0}').format(exit_code)
+
+    except ValueError as e:
+        exit_code = 11
+        message = (operation + ' failed with error: {0}').format(e.message)
+
+    except Exception as e:
+        exit_code = 1
+        message = (operation + ' failed with error: {0}, {1}').format(e, traceback.format_exc())
+
+        if "LINUXOMSAGENTEXTENSION_ERROR_MULTIPLECONNECTIONS" in str(e):
+            exit_code = 10
+            message = (operation + ' failed with error: {0}').format(e.message)
+
+    # Finish up and log messages
+    log_and_exit(hutil, operation, exit_code, message)
 
 
 def parse_context(operation):
     hutil = Util.HandlerUtility(waagent.Log, waagent.Error)
     hutil.do_parse_context(operation)
     return hutil
-    
-    
-def dummy_command(operation, status, msg):
-    hutil = parse_context(operation)
-    hutil.do_exit(0, operation, status, '0', msg)
 
 
-def getbundlefilename():
-    file_name = BundleFileNameTemplate.format('x86')
-    is_64bits = sys.maxsize > 2**32
-    if is_64bits:
-        file_name = BundleFileNameTemplate.format('x64')
-    return file_name
+def is_vm_supported_for_extension():
+    '''
+    Returns for platform.linux_distribution() vary widely in format, such as '7.3.1611' returned
+    for a machine with CentOS 7, so the first provided digits must match
```

Comment:
This is an interesting thing for us (LAD devs) to note as well. Thanks and later we may adopt your logic here in LAD as well.